webassembly/app.html

```diff
@@ -223,6 +223,54 @@ <h1 class="title">F3D Web</h1>
         // do a first render and start the interactor
         Module.engineInstance.getWindow().render();
         Module.engineInstance.getInteractor().start();
+
+        // Parse search-query mesh url-param or load default mesh file
+        const params = new URLSearchParams(window.location.search);
+        const mesh_url = params.get("mesh");
+        const extension = params.get("extension");
+        if (mesh_url) {
+          fetch(mesh_url)
+          .then((response) => {
+            if (!response.ok) {
+              throw new Error(`HTTP error, status = ${response.status}`);
+            }
+            // Build filename given extension urlparam or response header content-disposition
+            let filename 
+            if (extension) {
+              filename = `mesh_urlparam.${extension}`
+              console.log('filename constructed from user-provided extension', filename)
+            } else {
+              // If extension is not provided by user, try to get it auto from content-disposition header of url extension 
+              const contentDisposition = response.headers.get('content-disposition')
+              console.log('contentDisposition', contentDisposition)
+              if (contentDisposition) {
+                filename = contentDisposition.split('filename=')[1].split(';')[0];
+                console.log('filename constructed from contentDisposition header', filename)
+              } else {
+                try {
+                  filename = mesh_url.split('/').pop()
+                  console.log('filename constructed from mesh_url filename+extension', filename)
+                  const url_extension = '.' + mesh_url.split('.').at(-1) 
+                  if (url_extension.length > 10) {
+                    throw new Error(`Bad extension parsed from filename ${filename}`)
+                  }
```

Comment:
Is it necessary? If the extension is unknown, F3D will send an error message anyway.